pkg/storage/sqlstorage/migrate.go

```diff
@@ -3,21 +3,48 @@ package sqlstorage
 import (
 	"context"
 	"database/sql"
+	"sort"
+	"strconv"
 	"time"
 
 	"github.com/huandu/go-sqlbuilder"
 	"github.com/numary/go-libs/sharedlogging"
 )
 
+type HandlersByEngine map[string][]MigrationFunc
+
 type Migration struct {
 	Number   string
 	Name     string
 	Handlers HandlersByEngine
 }
 
+type Migrations []Migration
+
+func (m Migrations) Len() int {
+	return len(m)
+}
+
+func (m Migrations) Less(i, j int) bool {
+	iNumber, err := strconv.ParseInt(m[i].Number, 10, 64)
+	if err != nil {
+		panic(err)
+	}
+	jNumber, err := strconv.ParseInt(m[j].Number, 10, 64)
+	if err != nil {
+		panic(err)
+	}
+	return iNumber < jNumber
+}
+
+func (m Migrations) Swap(i, j int) {
+	m[i], m[j] = m[j], m[i]
+}
+
+var _ sort.Interface = &Migrations{}
```

Comment:
What is it for?

Reply:
@antoinegelloz Just a compile time assertion to assert *Migrations implements sort.Interface.